packages/config-manager/lib/index.js

```diff
@@ -0,0 +1,86 @@
+const deepFreeze = require("deep-freeze-strict");
+const { readFileSync } = require("fs");
+const { env } = require("process");
+const predefined = require("./predefined");
+
+function assertHexString(debugPath, string) {
+  if (!/^0x([0-9a-fA-F][0-9a-fA-F])*$/.test(string)) {
+    throw new Error(`${debugPath} must be a hex string!`);
+  }
+}
+
+function assertHash(debugPath, hash) {
+  assertHexString(debugPath, hash);
+  if (hash.length != 66) {
+    throw new Error(`${debugPath} must be a hex string of 66 bytes long!`);
+  }
+}
+
+function assertInteger(debugPath, i) {
+  if (i === "0x0") {
+    return;
+  }
+  if (!/^0x[1-9a-fA-F][0-9a-fA-F]*$/.test(i)) {
+    throw new Error(`${debugPath} must be a hex integer!`);
```

Comment:
Is `0x1` a valid hex integer ? It seems must be `0x01` ?

Reply:
We must use `0x1` here to represent a number, no leading zeros are allowed.